spec/controller/application_controller_spec.rb

```diff
@@ -0,0 +1,18 @@
+require "rails_helper"
+
+RSpec.describe ApplicationController, type: :controller do
+  describe "#request_ip" do
+    it "returns the anonymized IP with the last octet zero padded" do
+      expect(controller.request_ip).to eql("0.0.0.0")
+    end
+
+    context "when the IP is at the max range" do
+      it "returns the anonymized IP with the last octet zero padded" do
+        allow_any_instance_of(ActionController::TestRequest)
```

Comment:
This makes a bit more sense, I can see the setup and how the code has transformed the IP.